kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs

```diff
@@ -279,6 +279,191 @@ impl<'tcx> GotocCtx<'tcx> {
         }
     }
 
+    fn codegen_rvalue_enum(
+        &mut self,
+        variant_index: &VariantIdx,
+        operand_opt: Option<&Operand<'tcx>>,
+        ty: Ty<'tcx>,
+    ) -> Expr {
+        let overall_t = self.codegen_ty(ty);
+        let layout = self.layout_of(ty);
+        {
+            // TODO block of debug output, to be removed
+            debug!(?ty, "INPUT TYPE");
+            let components = overall_t.lookup_components(&self.symbol_table);
+            debug!(?components, "COMPONENTS");
+            assert!(components.is_some());
+            let union_type = components.unwrap()[components.unwrap().len() - 1].typ();
+            debug!(?union_type, "UNION TYPE");
+            // let full_union_type =
+            //     &self.symbol_table.lookup(union_type.type_name().unwrap()).unwrap().typ;
+            // debug!(?full_union_type, "FULL UNION TYPE");
+            debug!(?operand_opt);
+            // if it's an enum
+            // debug!(?variant_index);
+            debug!(?layout, "LAYOUT");
+        }
+        match &layout.variants {
+            Variants::Single { index } => {
+                if let ty::Adt(adt, subst) = &ty.kind() {
+                    debug!(?subst);
+                    // here we must have one variant
+                    let variant = &adt.variants()[*index];
+
+                    match variant.fields.len() {
+                        0 => {
+                            // would neither use the variant index nor the operand (if any),
+                            // no clear what this expression should mean
+                            unimplemented!()
+                            // Expr::struct_expr_from_values(overall_t, vec![], &self.symbol_table)
+                        }
+                        1 => {
+                            // let fty = variant.fields[0].ty(self.tcx, subst);
+                            // TODO: this is questionable - no cast, unconditional use of the
+                            // operand
+                            Expr::struct_expr_from_values(
+                                overall_t,
+                                vec![self.codegen_operand(operand_opt.unwrap())],
+                                &self.symbol_table,
+                            )
+                        }
+                        _ => unreachable!(),
```

Comment:
The variant itself can have many fields. Here you want to handle the assignment the same way you handled struct

Reply:
Done.